Divine/CLI/CommandLineArguments.cs

```diff
@@ -0,0 +1,327 @@
+            Description = "Set verbosity level of log output",
+            DefaultValue = "info",
+            AllowedValues = "info;warn;error;fatal;debug;silent",
+            ValueOptional = false,
+            Optional = true
+        )]
+        public string LogLevel;
+
+        // @formatter:off
+        [EnumeratedValueArgument(typeof(string), "gr2-options",
+            Description = "Set extra options for GR2/DAE conversion",
+            AllowMultiple = true,
+            AllowedValues = "export-normals;export-tangents;export-uvs;deduplicate-vertices;filter-uvs;recalculate-normals;recalculate-tangents;recalculate-iwt;flip-uvs;force-legacy-version;compact-tris;build-dummy-skeleton;apply-basis-transforms;conform",
+            ValueOptional = false,
+            Optional = true
+        )]
+        public string[] Options;
+
+        // @formatter:off
+        [EnumeratedValueArgument(typeof(string), 'o', "output-format",
+            Description = "Set output format for batch operations",
+            DefaultValue = null,
+            AllowedValues = "dae;gr2;lsv;pak;lsj;lsx;lsb;lsf",
+            ValueOptional = false,
+            Optional = true
+        )]
+        public string OutputFormat;
+
+        // @formatter:off
+        [EnumeratedValueArgument(typeof(string), 'p', "package-version",
+            Description = "Set package version",
+            DefaultValue = "v13",
+            AllowedValues = "v7;v9;v10;v13",
+            ValueOptional = false,
+            Optional = true
+        )]
+        public string PackageVersion;
+
+        // @formatter:off
+        [ValueArgument(typeof(string), 's', "source",
+            Description = "Set source file path or directory",
+            DefaultValue = null,
+            ValueOptional = false,
+            Optional = false
+        )]
+        public string Source;
+
+        // @formatter:on
+
+        public static LogLevel GetLogLevelByString(string optionLogLevel)
+        {
+            LogLevel logLevel;
+
+            switch (optionLogLevel)
+            {
+                case "off":
+                    logLevel = Enums.LogLevel.OFF;
+                    break;
+                case "fatal":
+                    logLevel = Enums.LogLevel.FATAL;
+                    break;
+                case "error":
+                    logLevel = Enums.LogLevel.ERROR;
+                    break;
+                case "warn":
+                    logLevel = Enums.LogLevel.WARN;
+                    break;
+                case "info":
+                    logLevel = Enums.LogLevel.INFO;
+                    break;
+                case "debug":
+                    logLevel = Enums.LogLevel.DEBUG;
+                    break;
+                case "trace":
+                    logLevel = Enums.LogLevel.TRACE;
+                    break;
+                case "all":
+                    logLevel = Enums.LogLevel.ALL;
+                    break;
+                default:

```

Comment:
In the Get*ByString functions we should probably throw an exception if an incorrect value is specified instead of returning an arbitrary default. (Or at least return the default only if the input is "empty").

Reply:
You can't pass an incorrect value. Argument values are constrained.